api/v1alpha1/wasm_types.go

```diff
@@ -81,8 +81,21 @@ type WasmCodeSource struct {
 
 	// SHA256 checksum that will be used to verify the wasm code.
 	//
+	// If not specified, Envoy Gateway will not verify the downloaded wasm code.
 	// kubebuilder:validation:Pattern=`^[a-f0-9]{64}$`
-	SHA256 string `json:"sha256"`
+	// +optional
+	SHA256 *string `json:"sha256"`
```

Comment:
this is tricky that it lives as top level, for image is this sha of the wasm module or the oci artifact

Reply:
Do you prefer having a SHA256 inside the HTTP and Wasm code source separately?